pkarr build with --no-default-features broken

```
cargo check -p pkarr --no-default-features
    Checking pkarr v2.0.1 (/home/dvc/pkarr/pkarr)
error[E0433]: failed to resolve: use of undeclared crate or module `rand`
 --> pkarr/src/keys.rs:5:5
  |
5 | use rand::rngs::OsRng;
  |     ^^^^ use of undeclared crate or module `rand`

warning: unused macro definition: `if_async`
  --> pkarr/src/lib.rs:38:14
   |
38 | macro_rules! if_async {
   |              ^^^^^^^^
   |
   = note: `#[warn(unused_macros)]` on by default

error[E0599]: no function or associated item named `generate` found for struct `SigningKey` in the current scope
   --> pkarr/src/keys.rs:17:51
    |
17  |         let signing_key: SigningKey = SigningKey::generate(&mut csprng);
    |                                                   ^^^^^^^^ function or associated item not found in `SigningKey`
    |
note: if you're trying to build a new `SigningKey` consider using one of the following associated functions:
      SigningKey::from_bytes
      SigningKey::from_keypair_bytes
   --> /home/dvc/.cargo/registry/src/index.crates.io-6f17d22bba15001f/ed25519-dalek-2.1.1/src/signing.rs:102:5
    |
102 |     pub fn from_bytes(secret_key: &SecretKey) -> Self {
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
...
136 |     pub fn from_keypair_bytes(bytes: &[u8; 64]) -> Result<SigningKey, SignatureError> {
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Some errors have detailed explanations: E0433, E0599.
For more information about an error, try `rustc --explain E0433`.
warning: `pkarr` (lib) generated 1 warning
error: could not compile `pkarr` (lib) due to 2 previous errors; 1 warning emitted
```

--- a/pkarr/src/keys.rs
+++ b/pkarr/src/keys.rs
@@ -2,6 +2,7 @@
 
 use crate::{Error, Result};
 use ed25519_dalek::{SecretKey, Signature, Signer, SigningKey, Verifier, VerifyingKey};
+#[cfg(feature = "rand")]
 use rand::rngs::OsRng;
 use std::{
     fmt::{self, Debug, Display, Formatter},
@@ -12,6 +13,7 @@ use std::{
 pub struct Keypair(SigningKey);
 
 impl Keypair {
+    #[cfg(feature = "rand")]
     pub fn random() -> Keypair {
         let mut csprng = OsRng;
         let signing_key: SigningKey = SigningKey::generate(&mut csprng);

